webapp/app/forms/fields.py

```diff
@@ -63,6 +63,46 @@ def _value(self):
         return self.data.split('-') if self.data else ''
 
 
+class IdNrWidget(MultipleInputFieldWidget):
+    """A divided input field with four text input fields, limited to two to three chars."""
+    sub_field_separator = ''
+    input_field_lengths = [2, 3, 3, 3]
+
+
+class IdNrField(SteuerlotseStringField):
+    def __init__(self, label='', validators=None, **kwargs):
+        super(IdNrField, self).__init__(label, validators, **kwargs)
+        self.widget = IdNrWidget()
+
+    def process_formdata(self, valuelist):
+        if valuelist:
+            self.data = valuelist
+        elif self.data is None:
+            self.data = []
+
+    def _value(self):
+        # In case the validation was not successful, we already have the data as a list of values.
+        if isinstance(self.data, list):
+            return self.data
+
+        # Once the validation has gone through, post_validate() stores the data as string.
```

Comment:
So _value is only called after post_validate has been called?

Reply:
Mh..no..the comments aren't clear enough, I see^^

Reply:
How about now?